cloud_images/centos7/dcos_vol_setup.sh

```diff
@@ -53,7 +53,12 @@ function main {
     echo "Adding entry to fstab: $fstab"
     echo "$fstab" >> /etc/fstab
     echo -n "Mounting: $device to $mount_location"
-    mount "$mount_location"
+    until grep ^$device /etc/mtab > /dev/null; do sleep 1; echo -n .; mount "$mount_location" || :; done
+    if [ "$mount_location" = "/var/log" ]; then
+      echo "Recreating /var/log/journal"
+      mkdir /var/log/journal
+      systemd-tmpfiles --create --prefix /var/log/journal
+    fi
```

Comment:
So the test I did on my local system just now (don't have access to the real thing atm.) was:

```
micromi:~ lukas$ time bash -c "echo foo; sleep 5" | :

real	0m0.004s
user	0m0.002s
sys	0m0.002s
micromi:~ lukas$ time bash -c "trap '' PIPE; echo foo; sleep 5" | :
bash: line 0: echo: write error: Broken pipe

real	0m5.013s
user	0m0.003s
sys	0m0.003s
```

The first one returns immediately, the second executes the `sleep`.